### ISSUE TITLE
import './invalid/path' should not be allowed

### ISSUE THREAD

**Issue by pocesar** [Microsoft/TypeScript#9917]
This is important because some libraries (like angular 1.x) have their own DI, and doesn't rely on imports / exports, and need to be imported for their registration of `angular.module('SomeModule', [])`

**TypeScript Version:**  nightly (2.1.0-dev.20160723)

**Code**

``` ts
// A *self-contained* demonstration of the problem follows...
import './invalid/path';
import * as doh from './path/exists';
import * as invalid from './invalid/path';
```

**Expected behavior:**

Should warn if the "just" import call is valid or invalid, like it happens with `import * as`

**Actual behavior:** 

No warning about `import './invalid/path'`, but does error for `import * as invalid from './invalid/path';` (`Cannot find module './invalid/path'`)

**Comment by mhegazy** [comment]
this is an intentional hole to allow for importing non-code resources, e.g. `json!./settings.json` (require), or `./settings.json!json` (system.js). the system only through an error if it fails to find the module if the type information from this module is needed, i.e. to know the type of an import that can used later on in the file.

**Comment by pocesar** [comment]
ah I see. a bit dangerous if you ask me. could we have a tsconfig switch? (I don't use "non" JS imports, but am using SystemJS)

**Comment by pocesar** [comment]
is this still relevant?
